### ISSUE TITLE
JSONPath Validation Prohibits Matching Nested Elements

### ISSUE THREAD

**Issue by pauldoppel** [mockoon/mockoon#1631]
**Describe the bug**
With Mockoon Version 9 JSONPath validation has been introduced for security reasons. The new regex-based validation in it's current state prohibits the matching of an object's child attributes.
The following screenshot from the [Mockoon JSONPath and object-path online evaluator](https://mockoon.com/tools/json-object-path-evaluator/) shows a simplified example of the selections I am trying to achieve:
<img width="1624" alt="Screenshot of json-object-path-evaluator" src="https://github.com/user-attachments/assets/194eb37e-18ec-44c8-87e6-6ec32a5ddb48" />

**Steps to Reproduce**
1. open and start the [invalid_jsonpath_mockoon_env.json](https://github.com/user-attachments/files/18179967/invalid_jsonpath_mockoon_env.json) Mockoon Environment that implements the above example
2. go to [http://localhost:3000](http://localhost:3000)
3. Find the output to be inconsistent with the expected results from above. Namely: 
<img width="1624" alt="Screenshot of the Bug's Occurence" src="https://github.com/user-attachments/assets/c1c222ad-630f-4e5d-b501-ee89f85aa616" />
4. to verify go to `http://localhost:3000/counter_example` and see that it works when omitting the matching operation from the JSONPath:
<img width="1624" alt="Screenshot of Successful JSONPath" src="https://github.com/user-attachments/assets/574a86b6-0532-4274-8459-e32daf93c366" />

I used the following command in a npm-based project with the latest `@mockoon/cli` installed to infer that this has indeed to do with the JSONPath validation, since the output is '`false`'
```
node -e 'const { isSafeJSONPath } = require("./node_modules/@mockoon/commons-server/dist/cjs/libs/utils.js");console.log(isSafeJSONPath("$[?(@.name.match(/.*?lex.*?/))]"))'
```

**Mockoon version:**
\>= 9.0 (as of Dec 18th 2024 9.0 and 9.1)

**OS / OS version:**
Reproduced on Windows 10 and MacOS Sequoia (15.1.1)

**Possible solutions:**
The simplest solution should be simply changing the `FILTER_RE` in [_packages/commons-server/src/libs/utils.ts_](https://github.com/mockoon/mockoon/blob/1ed31c4059d7f757f6cb2a43e10dc81b0d9c55a9/packages/commons-server/src/libs/utils.ts#L456) to include dots (`\.`) in the group preceding `*.match`. However, I am not nearly well informed enough about the vulnerabilities that were adressed with this implementation to realistically assess the dangers associated with doing it this way. 

For experimentation find the following version of an environment in regex101.com:
1. This version for the current implementation: [https://regex101.com/r/fmrc1l/1](https://regex101.com/r/fmrc1l/1)
2. This version for the proposed fix: [https://regex101.com/r/fmrc1l/2](https://regex101.com/r/fmrc1l/2)

**Comment by 255kb** [comment]
Thank you for opening such a detailed issue. Indeed, we whitelisted many syntaxes but not this one. 
In the meantime you can make it work like this: `$.[?(@property === "name" && @.match(/lex/i))]^`

**Comment by pauldoppel** [comment]
Thank you very much!

**Comment by 255kb** [comment]
📦 This was fixed in [v9.2.0](https://mockoon.com/releases/9.2.0/)
